pkg/app/csi-snapshot.go

```diff
@@ -0,0 +1,226 @@
+	return nil
+}
+
+func (csi *CSISnapshot) Initialize(ctx context.Context) error {
+	return nil
+}
+
+func (csi *CSISnapshot) Insert(ctx context.Context) error {
+	return nil
+}
+
+func (csi *CSISnapshot) Reset(ctx context.Context) error {
+	log.Print("Resetting the application.", field.M{"app": csi.name})
+
+	removeLogFileCmd := []string{"sh", "-c", "rm /var/log/time.log"}
+	stderr, err := csi.execCommand(ctx, removeLogFileCmd)
+	if err != nil {
+		return errors.Wrapf(err, "Error while deleting log file: %s", stderr)
+	}
+
+	log.Print("Reset of the application was successful.", field.M{"app": csi.name})
+	return nil
+}
+
+func (csi *CSISnapshot) IsReady(ctx context.Context) (bool, error) {
+	log.Print("Waiting for the application to be ready.", field.M{"app": csi.name})
+	ctx, cancel := context.WithTimeout(ctx, mssqlWaitTimeout)
+	defer cancel()
+
+	err := kube.WaitOnDeploymentReady(ctx, csi.cli, csi.namespace, csi.deployment.Name)
+	if err != nil {
+		return false, err
+	}
+	log.Print("Application instance is ready.", field.M{"app": csi.name})
+	return true, nil
+}
+
+func (csi *CSISnapshot) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "deployment",
+		Name:      csi.name,
+		Namespace: csi.namespace,
+	}
+}
+
+func (csi *CSISnapshot) Uninstall(ctx context.Context) error {
+	err := csi.cli.CoreV1().Namespaces().Delete(ctx, csi.namespace, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	log.Print("Namespace deleted successfully", field.M{"app": csi.name})
+	return nil
+}
+
+func (csi *CSISnapshot) Ping(ctx context.Context) error {
+	log.Print("Pinging the application.", field.M{"app": csi.name})
+
+	listDirectories := []string{"sh", "-c", "ls /var/log"}
+	stderr, err := csi.execCommand(ctx, listDirectories)
+	if err != nil {
+		return errors.Wrapf(err, "Error while Pinging the application %s", stderr)
+	}
+
+	log.Print("Ping to the application was success.", field.M{"app": csi.name})
+	return nil
+}
+
+func (csi *CSISnapshot) GetClusterScopedResources(ctx context.Context) []crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (csi *CSISnapshot) Count(ctx context.Context) (int, error) {
+	return 3, nil
+}
+
+func (csi *CSISnapshot) ConfigMaps() map[string]crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (csi *CSISnapshot) Secrets() map[string]crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (csi *CSISnapshot) execCommand(ctx context.Context, command []string) (string, error) {
+	podname, containername, err := kube.GetPodContainerFromDeployment(ctx, csi.cli, csi.namespace, csi.name)
+	if err != nil || podname == "" {
+		return "", errors.Wrapf(err, "Error getting pod and containername %s.", csi.name)
+	}
+	_, stderr, err := kube.Exec(csi.cli, csi.namespace, podname, containername, command, nil)
+	return stderr, err
+}
+
+func (csi CSISnapshot) getAppDeploymentObj() (*appsv1.Deployment, error) {
+	deploymentManifest := `apiVersion: apps/v1
```

Comment:
I think we should move these to Go structs. Maintaining raw strings is hard and is vulnerable to typos.